### PR TITLE
Add ppc32 to test PowerPC 440

### PR DESCRIPTION
**Pull request by shenki**

### CODE REVIEW

**Comment by shenki** [comment]
Opening a pull request for visibility and early feedback.

TODO:

- [ ] The guest is not able to cause qemu to exit. In the past we've used an expect script to look for a string that indicates success. Another option is to add this support to qemu

- [ ] Depends on https://github.com/ClangBuiltLinux/dockerimage/pull/12

**Comment by groeck** [comment]
> * [ ]  The guest is not able to cause qemu to exit. In the past we've used an expect script to look for a string that indicates success. Another option is to add this support to qemu

Looking into my scripts, I am able to use the reboot command together with "-no-reboot" to cause the VM to exit.

**Comment by shenki** [comment]
> > * [ ]   The guest is not able to cause qemu to exit. In the past we've used an expect script to look for a string that indicates success. Another option is to add this support to qemu
> 
> Looking into my scripts, I am able to use the reboot command in the guest together with "-no-reboot" to cause qemu to exit.

Nice! Thanks @groeck

**Comment by shenki** [comment]
Blocked on https://github.com/ClangBuiltLinux/dockerimage/pull/14

File: driver.sh [hunk 1]
```diff
@@ -51,6 +51,18 @@ setup_variables() {
...
 
+    "ppc32")
+      config=ppc44x_defconfig
+      image_name=zImage
+      qemu="qemu-system-ppc"
+      qemu_ram=128m
+      qemu_cmdline=( -machine bamboo
+                     -dtb "arch/powerpc/boot/dts/bamboo.dtb"
+                     -append "console=ttyS0"
+                     -initrd "images/ppc32/rootfs.cpio" )
```
Comment (nathanchance) [inline driver.sh]: Hmm shouldn't `-no-reboot` be added to this too?

Comment (shenki) [reply driver.sh]: Yes. I have it here locally... Good catch!

**Comment by shenki** [comment]
> Hopefully the patches in the -next folder can get picked up quickly so they make it into 4.21/5.0 and we > can get them backported and dropped from here.

I've been bugging the maintainer (who is in my team). They're on his radar and should make the merge window.

**Comment by shenki** [comment]
Running locally I just found a lockdep WARN_ON using this script: https://lore.kernel.org/linuxppc-dev/CACPK8XcEpPzBu28Uuz4hX8Xq2OM_LuAWD4Ljes+6k87YuQfYHw@mail.gmail.com/T/#u

**Comment by shenki** [comment]
The travis 'branch' CI was failing as I accidentally pushed the `ppc32` to the cbl repo instead of my fork, and then deleted it to correct my mistake.

**Pull request merged**